lib/bap_disasm/disasm.cpp

```diff
@@ -124,7 +124,7 @@ class disassembler {
         };
     }
 
-    void set_memory(int64_t addr, const char *data, int offset, int length) {
+    void set_memory(int64_t addr, const uint8_t *data, int offset, int length) {
```

Comment:
I don't think that they are really tied. Actually nothing that depends on LLVM should slip into this part of the library, as it should be independent from the actual backend. 

On the other hand, I'm not sure that it is a good idea to switch from `char *` to `uint8_t`. Because, it is not guaranteed by the standard that `unit8_t` is able to represent a characrter set, and that a byte has exactly 8 bits. So, semantically `char *` is more correct and a well accepted representation of the array.